addons/controllers/cpi/vspherecpiconfig_utils.go

```diff
@@ -377,22 +378,6 @@ func (r *VSphereCPIConfigReconciler) getOwnerCluster(ctx context.Context, cpiCon
 	return cluster, nil
 }
 
-// TODO: make these functions accessible to other controllers (for example csi) https://github.com/vmware-tanzu/tanzu-framework/issues/2086
-// getVSphereCluster gets the VSphereCluster CR for the cluster object
-func (r *VSphereCPIConfigReconciler) getVSphereCluster(ctx context.Context, cluster *clusterapiv1beta1.Cluster) (*capvv1beta1.VSphereCluster, error) {
-	vsphereCluster := &capvv1beta1.VSphereCluster{}
-	if err := r.Client.Get(ctx, types.NamespacedName{
-		Namespace: cluster.Namespace,
-		Name:      cluster.Name,
-	}, vsphereCluster); err != nil {
-		if apierrors.IsNotFound(err) {
-			return nil, errors.Errorf("VSphereCluster %s/%s not found", cluster.Namespace, cluster.Name)
-		}
-		return nil, errors.Errorf("VSphereCluster %s/%s could not be fetched, error %v", cluster.Namespace, cluster.Name, err)
-	}
-	return vsphereCluster, nil
-}
-
 // getSecret gets the secret object given its name and namespace
 func (r *VSphereCPIConfigReconciler) getSecret(ctx context.Context, namespace, name string) (*v1.Secret, error) {
```

Comment:
This line for fetching VsphereMachineTemplate needs the same fix. https://github.com/vmware-tanzu/tanzu-framework/blob/86a0e3d6dbd2b6ef1fccc006ae1d93f55e5f5cea/addons/controllers/cpi/vspherecpiconfig_utils.go#L62

Reply:
The same as above, we can get datacenter from the following variable:
```
- name: vcenter
      value:
        cloneMode: fullClone
        datacenter: /dc0
        datastore: /dc0/datastore/sharedVmfs-0
        folder: /dc0/vm
        network: /dc0/network/VM Network
        resourcePool: /dc0/host/cluster0/Resources
        server: 10.187.96.70
        storagePolicyID: ""
        template: /dc0/vm/photon-3-kube-v1.23.8+vmware.2-tkg.2-81d1a7892ad39f017fbaf59f9907cbe7
        tlsThumbprint: 1E:BB:09:45:FD:02:10:17:FD:55:8D:D1:05:8B:3F:A8:91:DD:29:0E
```